packages/flutter/test/material/search_test.dart

```diff
@@ -40,6 +40,26 @@ void main() {
     TestDefaultBinaryMessengerBinding.instance!.defaultBinaryMessenger.setMockMethodCallHandler(SystemChannels.platform, null);
   });
 
+  testWidgets('Changing query moves cursor to the end of query', (
+      WidgetTester tester) async {
+    final _TestSearchDelegate delegate = _TestSearchDelegate();
+
+    await tester.pumpWidget(TestHomePage(delegate: delegate));
+    await tester.tap(find.byTooltip('Search'));
+    await tester.pumpAndSettle();
```

Comment:
why the pump and settle?

Reply:
I took a reference from another test and I thought that would essentially complete all frames before I can actually set the query. Let me know if that needs to be changed.

Reply:
Generally we should avoid pumpAndSettle unless we're counting frames -- see the API docs.
Better to do `tester.pump()` followed by `tester.pump(const Duration(...));` with a duration that skips past the animations.

Reply:
@Hixie What do you think should be the ideal duration in subsequent `tester.pump()`?

Reply:
usually you want to put the duration for the animation.

Reply:
Yes, that is what I understood but I have no prior knowledge on what should be the correct duration there.

Reply:
it looks like it is a page transition of _SearchPageRoute, it is
```dart
const Duration(milliseconds: 300)
```

Reply:
generally i just put like 5 seconds or something like that that i know will be enough. what @chunhtai says is also valid.